pkg/controller/nodes/task/k8s/plugin_collector.go

```diff
@@ -0,0 +1,181 @@
+const KindKey contextutils.Key = "kind"
+
+// This object is responsible for emitting metrics that show the current number of a given K8s resource kind, cut by namespace.
+// It needs to be kicked off. The periodicity is not currently configurable because it seems unnecessary. It will also
+// a timer measuring how long it takes to run each measurement cycle.
+type ResourceLevelMonitor struct {
+	Scope promutils.Scope
+
+	// Meta timer - this times each collection cycle to measure how long it takes to collect the levels GaugeVec below
+	CollectorTimer *labeled.StopWatch
+
+	// System Observability: This is a labeled gauge that emits the current number of objects in the informer. It is used
+	// to monitor current levels.
+	Levels *labeled.Gauge
+
+	// This informer will be used to get a list of the underlying objects that we want a tally of
+	sharedInformer cache.SharedIndexInformer
+
+	// The kind here will be used to differentiate all the metrics, we'll leave out group and version for now
+	gvk schema.GroupVersionKind
+
+	// We only want to create one of these ResourceLevelMonitor objects per K8s resource type. If one of these objects
+	// already exists because another plugin created it, we pass along the pointer to that older object. However, we don't
+	// want the collection goroutine to be kicked off multiple times.
+	once sync.Once
+}
+
+// The reason that we use namespace as the one and only thing to cut by is because it's the feature that we are sure that any
+// K8s resource created by a plugin will have (as yet, Flyte doesn't have any plugins that create cluster level resources and
+// it probably won't for a long time). We can't assume that all the operators and CRDs that Flyte will ever work with will have
+// the exact same set of labels or annotations or owner references. The only thing we can really count on is namespace.
+func (r *ResourceLevelMonitor) countList(ctx context.Context, objects []interface{}) map[string]int {
+	// Map of namespace to counts
+	counts := map[string]int{}
+
+	// Collect the object counts by namespace
+	for _, v := range objects {
+		metadata, err := meta.Accessor(v)
+		if err != nil {
+			logger.Errorf(ctx, "Error converting obj %v to an Accessor %s\n", v, err)
+			continue
+		}
+		counts[metadata.GetNamespace()]++
+	}
+
+	return counts
+}
+
+// The context here is expected to already have a value for the KindKey
+func (r *ResourceLevelMonitor) collect(ctx context.Context) {
+	// Emit gauges at the namespace layer - since these are just K8s resources, we cannot be guaranteed to have the necessary
+	// information to derive project/domain
+	objects := r.sharedInformer.GetStore().List()
+	counts := r.countList(ctx, objects)
+
+	for ns, count := range counts {
+		withNamespaceCtx := contextutils.WithNamespace(ctx, ns)
+		r.Levels.Set(withNamespaceCtx, float64(count))
+	}
+}
+
+func (r *ResourceLevelMonitor) RunCollector(ctx context.Context) {
+	ticker := time.NewTicker(resourceLevelMonitorCycleDuration)
```

Comment:
defer ticker.Stop()

Reply:
you mean inside the goroutine right?  If I add it here, the ticker will just stop and it'll never run.

Reply:
added it inside.